### DISCUSSION TITLE
Error messages found with the new generic circuit feature

### DISCUSSION THREAD

**Discussion by lhcwha** [hneemann/Digital#628] [Q&A] [answered]
I am trying out the new generic circuit feature with custom component subcircuit. But I got two kinds of error message:
1. The code "_addComponent("My-ADD",10,10);_" (Test-001.dig) causes to "Component My-ADD not found" error message.
2. The code "_this.rotation=1;_" in generic parameterization field of My-OR custom component (Test-002.dig) causes to "Variable not found: this; line 1" error message.

Did I do something wrong? Or are these bugs?

[Test001.zip](https://github.com/hneemann/Digital/files/5888308/Test001.zip)

**Comment by hneemann** [comment, marked as answer]
1. If you want to add a custom component, you have to use the name including the ".dig": `addComponent("My-ADD.dig",10,10);`
2. The generic code section of custom components cannot change the component itself. Maybe I should add this, but it doesn't seem necessary. Moreover, it would cause problems because the code executed could overwrite itself. Instead, this section is used to create the parameterization of the embedded circuit itself.
If an embedded circuit needs parameters (which should always be the case since otherwise a generic circuit would have no sense), they are defined here. 
For example, if the parameter "args.bitWidth" is used in a generic circuit, it can be defined in the embedding component with the statement "bitWidth:=[value]".

**Reply by lhcwha** [reply]
Thank you very much for your kindly answer!
